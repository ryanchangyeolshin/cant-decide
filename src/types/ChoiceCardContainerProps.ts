type ChoiceCardContainerProps = {
  choice: string,
  index: number,
  removeChoice: Function,
}

export default ChoiceCardContainerProps;