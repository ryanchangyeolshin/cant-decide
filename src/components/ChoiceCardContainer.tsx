import React from 'react';
import { makeStyles, createStyles, Theme, } from '@material-ui/core/styles';
import {
  ListItem,
  ListItemIcon,
  ListItemText,
  ListItemSecondaryAction,
  IconButton,
  Badge,
  Grow
} from '@material-ui/core';
import DeleteIcon from '@material-ui/icons/Delete';
import ChoiceCardContainerProps from '../types/ChoiceCardContainerProps';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    listItem: {
      backgroundColor: "white",
      marginBottom: "10px",
      borderRadius: "4px",
      padding: "20px",
      transition: theme.transitions.create("background-color", {
        duration: 1000
      }),
      "&:hover": {
        backgroundColor: "yellow",
      },
      "& .MuiListItemText-root": {
        "& .MuiTypography-root.MuiListItemText-primary": {
          fontWeight: 900,
          overflow: "hidden",
          whiteSpace: "nowrap",
          textOverFlow: "ellipsis",
          width: "95%"
        }
      }
    }
  }),
);

const ChoiceCardContainer: React.FC<ChoiceCardContainerProps> = ({ choice, index, removeChoice }: ChoiceCardContainerProps) => {
  const classes = useStyles();

  const removeChoiceHandler = (e: React.MouseEvent<HTMLButtonElement>) => {
    removeChoice(index);
  } 

  return (
    <Grow in={true} timeout={1500} mountOnEnter unmountOnExit={true}>
      <div className="list-item-container">
        <ListItem button className={classes.listItem}>
          <ListItemIcon>
            <Badge badgeContent={index + 1} color="primary" />
          </ListItemIcon>
          <ListItemText primary={choice} />
          <ListItemSecondaryAction>
            <IconButton data-testid="delete-icon" edge="end" aria-label="delete" onClick={removeChoiceHandler}>
              <DeleteIcon />
            </IconButton>
          </ListItemSecondaryAction>
        </ListItem>
      </div>
    </Grow>
  );
}

export default ChoiceCardContainer;