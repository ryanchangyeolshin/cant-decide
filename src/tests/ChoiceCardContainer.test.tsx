import React, { useContext } from 'react';
import { render, getByTestId, fireEvent } from '@testing-library/react';
import ChoiceCardContainer from '../components/ChoiceCardContainer';
import ChoiceCardContainerProps from '../types/ChoiceCardContainerProps';

describe("This suit is to test the ChoiceCardContainer component", () => {
  const props: ChoiceCardContainerProps = {
    choice: "Taco Bell",
    index: 1,
    removeChoice: jest.fn(() => "hello"),
  };

  test('Snapshot of ChoiceCardContainer', () => {
    const { asFragment } = render(<ChoiceCardContainer choice={props.choice} index={props.index} removeChoice={props.removeChoice} />);
    expect(asFragment()).toMatchSnapshot()
  });

  test('that the choice text is rendered', async () => {
    const { getByText } = await render(<ChoiceCardContainer choice={props.choice} index={props.index} removeChoice={props.removeChoice} />);
    expect(getByText(props.choice)).toBeInTheDocument();
  });

  test('that the index badge is rendered', async () => {
    const { getByText } = await render(<ChoiceCardContainer choice={props.choice} index={props.index} removeChoice={props.removeChoice} />);
    expect(getByText(props.index + 1)).toBeInTheDocument();
  });

  test('that the delete trash icon is rendered', async () => {
    const { getByLabelText } = await render(<ChoiceCardContainer choice={props.choice} index={props.index} removeChoice={props.removeChoice} />);
    expect(getByLabelText("delete")).toBeInTheDocument();
  });

  test('that removeChoice function to be called after clicking the delete icon', async () => {
    const { container } = await render(<ChoiceCardContainer choice={props.choice} index={props.index} removeChoice={props.removeChoice} />);
    const deleteIcon = getByTestId(container, "delete-icon");
    fireEvent.click(deleteIcon);
    expect(props.removeChoice).toBeCalled();
  });
});
