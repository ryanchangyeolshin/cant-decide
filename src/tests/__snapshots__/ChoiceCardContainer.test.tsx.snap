// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This suit is to test the ChoiceCardContainer component Snapshot of ChoiceCardContainer 1`] = `
<DocumentFragment>
  <div
    class="list-item-container"
    style="opacity: 1; transform: scale(1, 1); transition: opacity 1500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 999ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  >
    <li
      class="MuiListItem-container"
    >
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root makeStyles-listItem-1 MuiListItem-gutters MuiListItem-button MuiListItem-secondaryAction"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root"
        >
          <span
            class="MuiBadge-root"
          >
            <span
              class="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorPrimary"
            >
              2
            </span>
          </span>
        </div>
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Taco Bell
          </span>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <div
        class="MuiListItemSecondaryAction-root"
      >
        <button
          aria-label="delete"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
          data-testid="delete-icon"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </li>
  </div>
</DocumentFragment>
`;
